File: api-openbis-java/source/java/ch/ethz/sis/openbis/generic/imagingapi/v3/dto/ImagingDataSetExportConfig.java
package ch.ethz.sis.openbis.generic.imagingapi.v3.dto;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.property.PropertiesDeserializer;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

@JsonObject("imaging.dto.ImagingDataSetExportConfig")
public class ImagingDataSetExportConfig implements Serializable
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private String archiveFormat;

    @JsonProperty
    private String imageFormat;

    @JsonProperty
    private String resolution;

    @JsonProperty
    private List<ImagingExportIncludeOptions> include = new ArrayList<>();

    @JsonProperty
    @JsonDeserialize(contentUsing = PropertiesDeserializer.class)
    private Map<String, String> customOptions;


    @JsonIgnore
    public String getArchiveFormat()
    {
        return archiveFormat;
    }

    public void setArchiveFormat(String archiveFormat)
    {
        this.archiveFormat = archiveFormat;
    }

    @JsonIgnore
    public String getImageFormat()
    {
        return imageFormat;
    }

    public void setImageFormat(String imageFormat)
    {
        this.imageFormat = imageFormat;
    }

    @JsonIgnore
    public String getResolution()
    {
        return resolution;
    }

    public void setResolution(String resolution)
    {
        this.resolution = resolution;
    }

    @JsonIgnore
    public List<ImagingExportIncludeOptions> getInclude()
    {
        return include;
    }

    public void setInclude(List<ImagingExportIncludeOptions> include)
    {
        this.include = include;
    }

    @JsonIgnore
    public Map<String, String> getCustomOptions()
    {
        return customOptions;
    }

    public void setCustomOptions(Map<String, String> customOptions)
    {
        this.customOptions = customOptions;
    }

}

File: api-openbis-javascript/src/v3/imaging/dto/ImagingDataSetExportConfig.js
define([ "stjs", "util/Exceptions" ], function(stjs, exceptions) {
	var ImagingDataSetExportConfig = function() {
	};
	stjs.extend(ImagingDataSetExportConfig, null, [], function(constructor, prototype) {
		prototype['@type'] = 'imaging.dto.ImagingDataSetExportConfig';
		constructor.serialVersionUID = 1;
		prototype.archiveFormat = null;
		prototype.imageFormat = null;
		prototype.resolution = null;
		prototype.include = null;
		prototype.customOptions = null;

		prototype.getArchiveFormat = function() {
            return this.archiveFormat;
        };
        prototype.setArchiveFormat = function(archiveFormat) {
            this.archiveFormat = archiveFormat;
        };
        prototype.getImageFormat = function() {
            return this.imageFormat;
        };
        prototype.setImageFormat = function(imageFormat) {
            this.imageFormat = imageFormat;
        };
		prototype.getResolution = function() {
			return this.resolution;
		};
		prototype.setResolution = function(resolution) {
			this.resolution = resolution;
		};
		prototype.getResolution = function() {
            return this.resolution;
        };
        prototype.setResolution = function(resolution) {
            this.resolution = resolution;
        };
        prototype.getInclude = function() {
            return this.include;
        };
        prototype.setInclude = function(include) {
            this.include = include;
        };
        prototype.getCustomOptions = function() {
            return this.customOptions;
        };
        prototype.setCustomOptions = function(customOptions) {
            this.customOptions = customOptions;
        };
		prototype.toString = function() {
            return "ImagingDataSetExportConfig: " + this.config;
        };

	}, {
		exports : {
            name : "List",
            arguments : [ "ImagingExportIncludeOptions"]
        },
        customOptions : {
            name : "Map",
            arguments : [ "String", "String" ]
        }
	});
	return ImagingDataSetExportConfig;
})